### PR TITLE
Smart build test

### PR DESCRIPTION
**Pull request by dtrawins**

### CODE REVIEW

File: rununittest.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ ${debug_bazel_flags} \
 --test_output=streamed \
 --test_env PYTHONPATH=${PYTHONPATH}"
 test_success_procedure() {
-    grep -a " ms \| ms)" ${TEST_LOG}
```
Comment (rasapala) [inline rununittest.sh:27]: Please leave this - shows times for tests and lists them nicely.

**Review by rasapala — APPROVED** [review]

File: ci/build_test_OnCommit.groovy [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        }
+
+        stage("Build docker image") {
+          when { expression { image_build_needed == "true" } }
+          steps {
+                sh "make ovms_builder_image RUN_TESTS=0 OV_USE_BINARY=1 OVMS_CPP_IMAGE_TAG=${shortCommit}"
+                sh "make release_image RUN_TESTS=0 OV_USE_BINARY=1 OVMS_CPP_IMAGE_TAG=${shortCommit}"
+              }
+        }
+
```
Comment (dkalinowski) [inline ci/build_test_OnCommit.groovy:50]: Missing tests for ovmsclient

File: ci/build_test_OnCommit.groovy [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+              echo sh(script: 'env|sort', returnStdout: true)
+              def git_diff = ""
+              if (env.CHANGE_ID){ // PR - check changes between target branch
+                sh 'git fetch origin ${CHANGE_TARGET}'
+                git_diff = sh (script: "git diff --name-only \$(git merge-base FETCH_HEAD HEAD)", returnStdout: true).trim()
+                println("git diff:\n${git_diff}")
+              } else {  // branches without PR - check changes in last commit
+                git_diff = sh (script: "git diff --name-only HEAD^..HEAD", returnStdout: true).trim()
+              }
+              def matched = (git_diff =~ /src|third_party|(\n|^)Dockerfile|(\n|^)Makefile|\.cpp|\.h|BUILD|WORKSPACE|(\n|^)rununittest\.sh/)
```
Comment (dkalinowski) [inline ci/build_test_OnCommit.groovy:23]:
```suggestion
              def matched = (git_diff =~ /src|third_party|(\n|^)Dockerfile|(\n|^)Makefile|\.c|\.h|BUILD|WORKSPACE|(\n|^)rununittest\.sh/)
```

**Review by ngrozae — APPROVED** [review]

File: ci/build_test_OnCommit.groovy [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+              echo sh(script: 'env|sort', returnStdout: true)
+              def git_diff = ""
+              if (env.CHANGE_ID){ // PR - check changes between target branch
+                sh 'git fetch origin ${CHANGE_TARGET}'
+                git_diff = sh (script: "git diff --name-only \$(git merge-base FETCH_HEAD HEAD)", returnStdout: true).trim()
+                println("git diff:\n${git_diff}")
+              } else {  // branches without PR - check changes in last commit
+                git_diff = sh (script: "git diff --name-only HEAD^..HEAD", returnStdout: true).trim()
+              }
+              def matched = (git_diff =~ /src|third_party|(\n|^)Dockerfile|(\n|^)Makefile|\.c|\.h|BUILD|WORKSPACE|(\n|^)rununittest\.sh/)
```
Comment (atobiszei) [inline ci/build_test_OnCommit.groovy:24]: This does not include all needed matches (bazel.rc, .bzl) files. Why we don't use blacklist instead?

**Review by atobiszei — CHANGES_REQUESTED** [review]

File: ci/build_test_OnCommit.groovy (hunk 4 above)
Comment (atobiszei) [reply ci/build_test_OnCommit.groovy:24]: Also rebuild will not trigger if you will add new licesne to `release_files`.

Comment (dtrawins) [reply ci/build_test_OnCommit.groovy:24]: @atobiszei  indeed but the license file has no impact on the image behavior and oncommit testing results. In my opinion there is not point in building the image and functional tests only if the license changes

**Review by atobiszei — APPROVED** [review]

File: ci/build_test_OnCommit.groovy [hunk 5]
```diff
@@ -21,7 +21,7 @@ pipeline {
               } else {  // branches without PR - check changes in last commit
                 git_diff = sh (script: "git diff --name-only HEAD^..HEAD", returnStdout: true).trim()
               }
-              def matched = (git_diff =~ /src|third_party|(\n|^)Dockerfile|(\n|^)Makefile|\.c|\.h|BUILD|WORKSPACE|(\n|^)rununittest\.sh/)
```
Comment (dkalinowski) [inline ci/build_test_OnCommit.groovy]: Why are we removing "BUILD"? There are files called "BUILD" that should require image rebuild.
- BUILD
- BUILD.bazel

**Pull request merged**